### PR TITLE
added-decorators

### PR DESCRIPTION
**Pull request by AhmedNazirMusah**
On this milestone I implemented the following:
- Created a class Nameable.
Implemented a method called correct_name that will raise a NotImplementedError.
- Turned the Person class to Nameable
- Person class inherits from Nameable and this class has a method correct_name implemented. It should simply return the name attribute.
- Prepared a base Decorator and it inherits from Nameable.

- Implemented the correct_name method that returns the result of the correct_name method of the @nameable.
- Prepared CapitalizeDecorator and TrimmerDecorator
The CapitalizeDecorator has a method correct_name which capitalizes the output of @nameable.correct_name.
The TrimmerDecorator has a method correct_name which makes sure that the output of @nameable.correct_name has a maximum of 10 characters. If it's longer it should trim the word.

### CODE REVIEW

File: classes/base_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+require_relative 'nameable'
+
+# class Decorator
+class Decorator < Nameable
```
Comment (BenMukebo) [inline classes/base_decorator.rb:4]: - [ ] Getters and Setters for `nameable` should be present to allow us to access and update the `:nameable` from outside the scope of the class definition. Kindly add this 👇🏻 

`attr_accessor :nameable`

File: classes/trimmer_decorator.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+require_relative 'base_decorator'
+
+# class  trimmer decorator
+class TrimmerDecorator < Decorator
+  def correct_name
+    @nameable.correct_name[0...10].to_s
```
Comment (BenMukebo) [inline classes/trimmer_decorator.rb]: - [OPTIONAL] You can also test If your name is long and has a maximum of 10 before you Trimmer it by adding a test length condition like this 👇🏻 

```ruby
    def correct_name
      @nameable.correct_name.length > 10 ? @nameable.correct_name[0..9] : @nameable.correct_name
    end
```

File: classes/person.rb [hunk 3]
```diff
@@ -22,3 +31,15 @@ def of_age
...
+
+puts person.correct_name
+
+capitalized_person = CapitalizeDecorator.new(person)
+
+puts capitalized_person.correct_name
+
+capitalized_trimmed_person = TrimmerDecorator.new(capitalized_person)
+
+puts capitalized_trimmed_person.correct_name
```
Comment (BenMukebo) [inline classes/person.rb:45]: - Great job here 💯 😉 👍🏻

**Review by BenMukebo — APPROVED** [review]
<h1>Status: Approve  :champagne:  :+1: :confetti_ball: 

### **Hi @AhmedNazirMusah**, 

### **Congratulations! 🎉 Project Approved ✅**

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 


### To Highlight :1st_place_medal: 
- Your Pull Request has a proper title and description. ✔️  
- No Linter errors :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


- [OPTIONAL] I would also recommend adding the issue link in the contributing sections: 

```
## 🤝 Contributing

Contributions, issues, and feature requests are welcome!

Feel free to check the [issues page](../../issues/).
```

Cheers :clinking_glasses:  and Happy coding!!! :dancers:💻  

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
